### ISSUE TITLE
Fn::ToJsonString: many larger companies do not feel comfortable using transforms

### ISSUE THREAD

**Issue by rix0rrr** [aws-cloudformation/cfn-language-discussion#105]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### **Tell us about your request**

It would be helpful if `Fn::ToJsonString` was a "proper" intrinsic, instead of provided by a transform. This would have two benefits:

- We could properly JSONify `{Fn::GetAtt}`s which are anything other than strings (numbers, arrays, booleans, ...)
- The intrinsic could be used in places where CloudFormation transforms are banned

We have spoken to a good number of customers during the investigation of a different feature which we were going to implement as a transform, and a surprising amount of them said that transforms were banned at their company, for fear of unpredictability ("no way to know what the transform would do").

(Before shooting the messenger: yes, I personally am aware of `ProcessedTemplate`, and I personally also do think those concerns are unnecessary. However, they are what they are, and we could hit 2 birds with one stone by making these functions proper intrinsics).

### **Tell us about the problem you are trying to solve. What are you trying to do, and why is it hard?**

I'm from the CDK team. It would make our life A LOT simpler if we could rely on CloudFormation built-ins to produce JSON strings from object literals (given that 99.9% of AWS APIs run on JSON 😅).

However, as long as `Fn::ToJsonString` requires a mechanism that is banned by *any company*, we cannot switch over to it.

### **Are you currently working around this issue?**

We are currently *guessing* at some JSONification of strings, and provisioning a Custom Resource Lambda into our customers' accounts if it looks like we'll need to JSONify arrays.

Customers also hate the Lambdas we put into their accounts for CFN deployment purposes, but we currently have no workaround.

### **What is the expect behavior with this new feature**

The same as it does today (well, covering more of JSON and potentially even going beyond the standard to also encode top-level scalars), except without requiring a transform.

**Comment by MalikAtalla-AWS** [comment]
Hey @rix0rrr, thanks for raising this issue. I understand where you're coming from. I'd like to gauge the customer interest for your request. Maybe we can tweak the title to make it more discoverable and potentially see some :+1: or comments on this issue. That would help us prioritize. How about I change the title to "Allow using Fn::ToJsonString without Transform"?

**Comment by ronkorving** [comment]
Recently, the Serverless Framework [started supporting](https://github.com/serverless/serverless/releases/tag/v3.24.0) the use of `Fn::ToJsonString`, so more people will likely start using it there as well (I'm one of them).

**Comment by zxkane** [comment]
Language extensions is not supported by all regions. It's an another drawback to adapt it.

**Comment by MalikAtalla-AWS** [comment]
Hi @zxkane, I understand the concern. AWS::LanguageExtensions recently expanded into [5 more AWS regions](https://aws.amazon.com/about-aws/whats-new/2023/03/aws-cloudformation-language-extensions-transform-5-regions/) and is now available in all public AWS regions. We are also working on making sure that AWS::LanguageExtensions will be immediately available whenever a new AWS region launches in the future.

**Comment by zxkane** [comment]
@MalikAtalla-AWS will AWS China regions be included too?

**Comment by MalikAtalla-AWS** [comment]
Yes, @zxkane, AWS::LanguageExtensions is also available in the AWS China regions China (Beijing) and China (Ningxia).
